### ISSUE TITLE
Incorrect distro detection in distros.nim

### ISSUE THREAD

**Issue by PrimordialHelios** [nim-lang/Nim#10036]
`detectOs(Fedora)` returns false on Fedora 29

### Example
```nim
import distros
echo detectOs(Linux)
echo detectOs(Posix)
echo detectOs(Fedora)
```

### Current Output
```
true
true
false
```

### Expected Output
```
true
true
true
```

### Possible Solution
Looks like the trouble starts with [this line](https://github.com/nim-lang/Nim/blob/master/lib/pure/distros.nim#L173). The `detectOsImpl` proc is looking for "fedora" in the uname string, but at least on my system, `uname -a` returns `Linux localhost.localdomain 4.19.8-300.fc29.x86_64 #1 SMP Mon Dec 10 15:23:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`, and `uname -o` returns `GNU/Linux`.

A possible fix is checking if the file `/etc/fedora-release` exists, or check `/etc/os-release` (the latter on my system symlinks to `/usr/lib/os-release`).

### Additional Information
My server running Fedora 27 also has the same issue, so this doesn't seem to be a problem only with Fedora 29.

**Comment by Araq** [comment]
Ok, thank you. Can you please create a PR for this? I can't test it, cannot have every distro under the sun to test these things.

**Comment by PrimordialHelios** [comment]
How would you prefer this be done? I can check for /etc/fedora-release, read and check OS name in /etc/os-release, or I can just `execProcess "hostnamectl"` and check that for the OS name.

I personally think the `hostnamectl` solution would be the best since it isn't trying to work with files at all.

**Comment by Araq** [comment]
I prefer the file check solution, ``execProcess`` is not avialable for Nimscript.

**Comment by PrimordialHelios** [comment]
All of the other distros are calling either `uname` or `release`, both of which depend use `execProcess`. I figured it would be best to follow their example. I've pushed a commit [here](https://github.com/PrimordialHelios/Nim/commit/835d45729184e0230851a41900d3bedf364c3a2f) that shows what I mean.

**Comment by Araq** [comment]
Ok, I misremembered, your solution is perfectly fine.

**Comment by PrimordialHelios** [comment]
Great! I'll send the PR

**Comment by PrimordialHelios** [comment]
Fixed in #10048
